packages/java/parser-jvm-plugin-transfertypes/src/main/java/com/vaadin/hilla/parser/plugins/transfertypes/TransferTypesPlugin.java

```diff
@@ -50,18 +55,59 @@ public final class TransferTypesPlugin
         classMap.put(ArrayNode.class.getName(), List.class);
         classMap.put("org.springframework.web.multipart.MultipartFile",
                 File.class);
+        classMap.put("com.vaadin.hilla.signals.Signal", Signal.class);
+        classMap.put("com.vaadin.hilla.signals.ValueSignal", ValueSignal.class);
+        classMap.put("com.vaadin.hilla.signals.NumberSignal",
+                NumberSignal.class);
+        classMap.put("com.vaadin.hilla.signals.ListSignal", ListSignal.class);
     }
 
-    public TransferTypesPlugin() {
-        super();
-    }
+    private final Set<String> replacedClasses = new HashSet<>();
 
     @Override
     public void enter(NodePath<?> nodePath) {
     }
 
     @Override
     public void exit(NodePath<?> nodePath) {
+        if (nodePath.getNode() instanceof EntityNode entityNode && nodePath
+                .getParentPath().getNode() instanceof RootNode rootNode) {
+            var cls = entityNode.getSource();
+            var name = cls.getName();
+            if (replacedClasses.contains(name)) {
+                var schema = entityNode.getTarget();
+
+                cls.getAnnotations().stream()
+                        .filter((model) -> model.getName()
+                                .equals(FromModule.class.getName()))
+                        .findFirst().ifPresent((annotationModel) -> {
+                            var annotation = (FromModule) annotationModel.get();
+                            var namedSpecifier = annotation.namedSpecifier();
+                            var defaultSpecifier = annotation
+                                    .defaultSpecifier();
+
+                            if (namedSpecifier.isBlank()
+                                    && defaultSpecifier.isBlank()) {
+                                throw new IllegalArgumentException(String
+                                        .format("@FromImport annotation for class %s must have at least one named specifier or a default specifier",
```

Comment:
I assume this is typo:
```suggestion
                                        .format("@FromModule annotation for class %s must have at least one named specifier or a default specifier",
```